Core/include/Acts/Seeding/PathSeeder.hpp

```diff
@@ -0,0 +1,185 @@
+    TrackEstimator trackEstimator;
+    /// Surface accessor
+    SourceLinkCalibrator sourceLinkCalibrator;
+    /// Intersection finder
+    IntersectionLookup intersectionFinder;
+    /// Path width provider
+    PathWidthLookup pathWidthProvider;
+    /// First layer extent
+    Extent firstLayerExtent;
+    /// Direction of the telescope extent
+    BinningValue orientation = BinningValue::binX;
+  };
+
+  /// @brief Constructor
+  PathSeeder(const Config& config) : m_cfg(std::move(config)){};
+
+  /// @brief Destructor
+  ~PathSeeder() = default;
+
+  /// @brief Extract the IP parameters and
+  /// sort the source links into the seeds
+  ///
+  /// @param gctx The geometry context
+  /// @param sourceLinks The source links to seed
+  ///
+  /// @return The vector of seeds
+  std::vector<Seed> getSeeds(const GeometryContext& gctx,
+                             const std::vector<SourceLink>& sourceLinks) const {
+    // Sort the source links into the grid
+    m_cfg.sourceLinkGrid->initialize(gctx, sourceLinks);
+
+    // Get plane of the telescope
+    // sensitive surfaces
+    int bin0 = static_cast<int>(BinningValue::binX);
+    int bin1 = static_cast<int>(BinningValue::binY);
+    if (m_cfg.orientation == BinningValue::binX) {
+      bin0 = static_cast<int>(BinningValue::binY);
+      bin1 = static_cast<int>(BinningValue::binZ);
+    } else if (m_cfg.orientation == BinningValue::binY) {
+      bin0 = static_cast<int>(BinningValue::binX);
+      bin1 = static_cast<int>(BinningValue::binZ);
+    }
+
+    // Create the seeds
+    std::vector<Seed> seeds;
+    for (const auto& sl : sourceLinks) {
+      Vector3 globalPos = m_cfg.sourceLinkCalibrator(gctx, sl);
+
+      // Check if the hit is in the
+      // first tracking layer
+      if (!m_cfg.firstLayerExtent.contains(globalPos)) {
+        continue;
+      }
+
+      // Get the IP parameters
+      auto [ipP, ipVertex, ipDir, flDir] =
+          m_cfg.trackEstimator(gctx, globalPos);
+
+      // Intersect with the surfaces
+      std::vector<std::pair<GeometryIdentifier, Vector3>> intersections =
+          m_cfg.intersectionFinder(gctx, globalPos, flDir, ipP);
+
+      // Continue if no intersections
+      if (intersections.empty()) {
+        continue;
+      }
+      // Vector to store the source links
+      std::vector<SourceLink> seedSourceLinks;
+
+      // Store the pivot source link
+      seedSourceLinks.push_back(sl);
+
+      // Iterate over the intersections
+      // and get the source links
+      // in the subsequent layers
+      for (auto& [geoId, refPoint] : intersections) {
+        // Get the path width
+        auto [pathWidth0, pathWidth1] = m_cfg.pathWidthProvider(gctx, geoId);
+
+        // Get the bounds of the path
+        ActsScalar top0 = refPoint[bin0] + pathWidth0;
+        ActsScalar bot0 = refPoint[bin0] - pathWidth0;
+        ActsScalar top1 = refPoint[bin1] + pathWidth1;
+        ActsScalar bot1 = refPoint[bin1] - pathWidth1;
```

Comment:
We do have a `Frustum` implementation that should give you the intersection points in an opening angle, again, this could make this code more general (but probably a bit slower).

Reply:
I looked into it. Doing something like that makes sense if we want to be able to deal with non-rectangular shapes. 

I wasn't planning to generalize the code to that extent but if you think that's a direction we should take, I would suggest instead merging the `pathWidthProvider` and the `intersectionFinder` functionality, so that the latter provides the rectangular bounds (in an arbitrary coordinate space) to be used in the grid search.  

That way the whole telescope business gets abstracted away and the user is free to choose in which binning convention they would like to work (Given that we would replace the `orientaion` with the `BinningValue` pair, or something like that).

Reply:
Decided to settle on just the telescope case